### PR TITLE
Replace timer with ticker in download progress reporter

### PR DESCRIPTION
**Pull request by ycombinator**
## What does this PR do?

This PR fixes a bug in the progress logger being used while downloading the Agent artifact during the upgrade process.  Prior to this PR the progress logger used a `time.Timer` to log progress of the download.  With this PR, the progress logger uses a `time.Ticker` instead.

## Why is it important?

The progress logger should _periodically_ log progress of the download, not just once (after a delay).

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] ~I have made corresponding changes to the documentation~
- [ ] ~I have made corresponding change to the default configuration files~
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)
- [ ] ~I have added an integration test or an E2E test~

## Related issues

- Found while writing tests for https://github.com/elastic/elastic-agent/pull/3542

### CODE REVIEW

File: internal/pkg/agent/application/upgrade/artifact/download/http/downloader.go [hunk 1]
```diff
@@ -223,7 +227,6 @@ func (e *Downloader) downloadFile(ctx context.Context, artifactName, filename, f
 type downloadProgressReporter struct {
 	log         progressLogger
 	sourceURI   string
-	timeout     time.Duration
```
Comment (ycombinator) [inline internal/pkg/agent/application/upgrade/artifact/download/http/downloader.go:226]: This change is unrelated to this PR _per se_.  But I noticed while working on this PR that this field was never being used, so I'm deleting it as part of this PR.

File: internal/pkg/agent/application/upgrade/artifact/download/http/downloader_test.go [hunk 2]
```diff
@@ -75,11 +75,11 @@ func TestDownloadBodyError(t *testing.T) {
 }
 
 func TestDownloadLogProgressWithLength(t *testing.T) {
-	fileSize := 100 * units.MiB
+	fileSize := 100 * units.MB
```
Comment (ycombinator) [inline internal/pkg/agent/application/upgrade/artifact/download/http/downloader_test.go:78]: This is just to make the math easier.

**Review by pchila — APPROVED** [review]
LGTM

**Review by michalpristas — APPROVED** [review]
